src/eduid_common/authn/vccs.py

```diff
@@ -130,21 +130,21 @@ def add_credentials(vccs_url, old_password, new_password,
 
     if old_factor:
         vccs.revoke_credentials(str(user.user_id), [old_factor])
-        user.passwords.remove(checked_password.id)
+        user.credentials.remove(checked_password.credential_id)
         logger.debug("Revoked old credential {!s} (user {!s})".format(
             old_factor.credential_id, user))
 
     if not old_password_supplied:
         # TODO: Revoke all current credentials on password reset for now
         revoked = []
-        for password in user.passwords.to_list():
-            revoked.append(vccs_client.VCCSRevokeFactor(str(password.id),
+        for password in user.credentials.filter(Password).to_list():
+            revoked.append(vccs_client.VCCSRevokeFactor(str(password.credential_id),
                                                         'reset password',
                                                         reference=source))
             logger.debug("Revoking old credential (password reset) "
-                      "{!s} (user {!s})".format(
-                          password.id, user))
-            user.passwords.remove(password.id)
+                         "{!s} (user {!s})".format(
+                          password.credential_id, user))
+            user.credentials.remove(password.credential_id)
```

Comment:
this one should also remove(password.key) but the logging above should arguably continue logging .credential_id and not the list reference .key